### ISSUE TITLE
Thanks a lot Mike! and I've a question

### ISSUE THREAD

**Issue by itsMasoud** [therealmagicmike/PBKDF2.NET#3]
Hi Mike,
Thank you so much for sharing this library! it's very useful for me.
i've used this (https://github.com/defuse/password-hashing) before to make a new instance of **PBKDF2**.
but it's implemented for **SHA1** hash algorithm. and i want **SHA512**.
Your implementation is so useful and good for this reason.
but i've find something that i don't no it's true or not?
when i used both algorithms in same page, password, salt size, but different hash algorithms **(SHA512 vs. SHA1)**.
as you know **SHA512** is slower than **SHA1**. but i've saw that your algorithm is faster than defuse's.
i want to know am i right or i'm in a wrong way? and, if i'm wrong, what should i do to fix this problem?
Thank you so much, Masoud.

**Comment by therealmagicmike** [comment]
Hi and thanks for the kind words!

At a quick glance I can see a few things that would cause this:

The other repo has its default iterations set to 64000, where as this implementation has a default of 1000 to support down-level hardware. To correct this for your usage, increase the iteration count to be used.

Another step that would effect speed comparisons is that the other repo is base64-encoding the output and building up a formatted string to return the results. I chose not to do this directly in my library as I wanted to leave the final storage format to the discretion of the user.

The most prominent difference you are probably seeing is directly in the comparison to verify two hashes against each other. The other repo implements a slow-compare function to compare the results where as this library does not provide a compare function, so I'm guessing you are comparing the to arrays in a standard equality check which would blaze through pretty fast. I chose not to implement a slow compare function as it's not part of the standard and would be more of a utility function than part of a PBKDF2 algorithm.

It's been a while since I revisited this project as I've been busy with other projects but I'm  considering an update as well as more current versions and I'll definitely consider adding some utility classes to further simplify user-implementations.
